src/main/java/org/tikv/common/TiSession.java

```diff
@@ -117,11 +118,21 @@ private synchronized void warmUp() {
             null,
             new TiStore(this.client.getStore(ConcreteBackOffer.newGetBackOff(), store.getId())));
       }
-      ByteString startKey = ByteString.EMPTY;
 
+      // use scan region to load region cache with limit
+      ByteString startKey = ByteString.EMPTY;
       do {
-        TiRegion region = regionManager.getRegionByKey(startKey);
-        startKey = region.getEndKey();
+        List<Pdpb.Region> regions =
+            regionManager.scanRegions(
+                ConcreteBackOffer.newGetBackOff(),
+                startKey,
+                ByteString.EMPTY,
+                conf.getScanRegionsLimit());
+        for (Pdpb.Region region : regions) {
+          regionManager.insertRegionToCache(
+              regionManager.createRegion(region.getRegion(), ConcreteBackOffer.newGetBackOff()));
+        }
+        startKey = regions.get(regions.size() - 1).getRegion().getEndKey();
       } while (!startKey.isEmpty());
 
       RawKVClient rawKVClient = createRawClient();
```

Comment:
should we close the client after the warmup is finished?

Reply:
done